eva-accession-release-automation/include_mapping_weight_from_dbsnp/incorporate_mapping_weight_into_accessioning.py

```diff
@@ -0,0 +1,157 @@
+        logger.info("Preparing update statement {0} for collection {1}".format(update_statement.__repr__(),
+                                                                               collection_name))
+        update_statements.append(update_statement)
+    return update_statements
+
+
+def bulk_update(collection_handle, update_statements, batch_size=1000, force_update=False):
+    if force_update or len(update_statements) >= batch_size:
+        try:
+            collection_handle.bulk_write(requests=update_statements, ordered=False)
+            return True
+        except Exception as ex:
+            logger.error(ex)
+            logger.error(traceback.format_exc())
+            sys.exit(1)
+    return False
+
+
+def incorporate_mapping_weight_for_assembly(accessioning_mongo_handle: pymongo.MongoClient,
+                                            metadata_connection_handle: pymongo.MongoClient,
+                                            GCA_accession: str):
+    accessioning_database_handle = accessioning_mongo_handle[eva_accession_database]
+
+    for schema_name, table_name, assembly in get_snpmapinfo_tables_with_GCA_assembly(metadata_connection_handle,
+                                                                                     GCA_accession):
+        species_info = get_species_info(metadata_connection_handle, schema_name)[0]
+        query_to_get_mapping_weight = get_mapping_weight_query(assembly, schema_name, table_name)
+        logger.info("Running query to get mapping weight: " + query_to_get_mapping_weight)
+        with get_db_conn_for_species(species_info) as species_connection_handle, \
+                get_result_cursor(species_connection_handle, query_to_get_mapping_weight) as cursor:
+            for result in cursor:
+                snp_id, weight = result[0], result[1]
+                for collection_name, collection_attributes in collections_to_update.items():
+                    collection_update_statements = collection_attributes["update_statements"]
+                    collection_handle = accessioning_database_handle[collection_name]
+                    collection_update_statements.extend(
+                        get_collection_update_statements(
+                            collection_name,
+                            collection_handle,
+                            collection_attributes["assembly_attribute_name"], GCA_accession,
+                            collection_attributes["rs_accession_attribute_name"], snp_id,
+                            collection_attributes["mapping_weight_attribute_path"], weight)
+                    )
+                    if bulk_update(collection_handle, collection_update_statements):
+                        collection_update_statements = []
+                    collections_to_update[collection_name]["update_statements"] = collection_update_statements
```

Comment:
Also is it more efficient to bulk statement on one collection together or would mixing statement across collections together makes little difference?

Reply:
At any given time **bulk_update** can perform updates on only one collection (hence the **collection_handle** parameter)

Reply:
> I think it would make for an easier read if the statements were produced by generator and a grouper function

Since this script will probably be run only once, I have dispensed with the more complicated control structures here but I have incorporated your other suggestion.